### PR TITLE
Encd 6225 valis track not refreshed

### PR DESCRIPTION
**Pull request by zoldello**

### CODE REVIEW

File: src/encoded/static/components/app.js [hunk 1]
```diff
@@ -322,8 +321,10 @@ class App extends React.Component {
             logout: 'triggerLogout',
         };
 
+        this.href = props.href;
```
Comment (emro) [inline src/encoded/static/components/app.js:323]: Since we are using `this.href` here, do we need `this.state.href` set in line 301?

**Review by emro — DISMISSED** [review]
Great solution! Just have one comment - I think we might be able to get rid of a line where this.state.href is set since we're not updating it and since we're hopefully using this.href everywhere.

**Review by emro — DISMISSED** [review]
Looks great!

**Comment by zoldello** [comment]
Added @forresttanaka to code review the new commit as he has the most knowledge about the issue I looked at here

**Review by forresttanaka — DISMISSED** [review]
Glad you got it working. Do you know what this.contextRequest is for? I still don’t even know.

File: src/encoded/static/components/form.js [hunk 2]
```diff
@@ -890,6 +891,13 @@ export class Form extends React.Component {
         return this.state.isDirty && this.state.isValid && !this.state.editor_error && !this.communicating;
     }
 
+    cancel() {
+        const link = url.parse(this.context.location_href, true);
+
+        // the last '/' is a hack to reload the page. In app.js, fallbackNavigate() did not without it.
+        this.context.navigate(`${link.pathname}/`, { reload: true });
```
Comment (forresttanaka) [inline src/encoded/static/components/form.js:899]: Thanks for that comment. It looks wrong to add the slash without it.

File: src/encoded/static/components/form.js [hunk 3]
```diff
@@ -890,6 +891,13 @@ export class Form extends React.Component {
...
+    }
```
Comment (forresttanaka) [inline src/encoded/static/components/form.js:900]: Thought you had to bind `this` for `cancel()` but now see you’re calling it in an arrow function.

**Review by forresttanaka — DISMISSED** [review]
Hopefully that’s all the special cases. That `<App>` change was pretty impactful, but I think we’re much better for it.

File: src/encoded/static/components/form.js (hunk 2 above)
Comment (zoldello) [reply src/encoded/static/components/form.js:899]: In fallbackNavigate() (in app.js), window.location.assign would not redirect properly without it. I have no idea why. That is why I did this. I also did not want to add too much internal details about app.js in form.js and other files.

File: src/encoded/static/components/form.js (hunk 3 above)
Comment (zoldello) [reply src/encoded/static/components/form.js:900]: Opps. You are correct. I will correct it. I have been dealing hooks for too long.

**Review by forresttanaka — APPROVED** [review]

**Pull request merged**